### PR TITLE
fix: [CDS-71737]: pipeline webook notification support

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots
after
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vikrant-harness — DISMISSED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureMSTeamsNotifications/ConfigureMSTeamsNotifications.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ import css from '@rbac/modals/ConfigureNotificationsModal/ConfigureNotifications
 
 interface MSTeamsNotificationsData {
   msTeamKeys: string[]
-  userGroups: string[]
+  userGroups?: string[]
```
Comment (AshwinBhatkal-Harness) [inline src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureMSTeamsNotifications/ConfigureMSTeamsNotifications.tsx:40]: Why is it made optional?

**Review by bhavya-sinha — DISMISSED** [review]

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger cypress-cv

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/ConfigureWebhookNotifications.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  submitButtonText?: string
+  config?: WebhookNotificationConfiguration
+  expressions?: string[]
+}
+
+interface WebhookNotificationData {
+  webhookUrl: string
+}
+
+export const TestWebhookNotifications: React.FC<{
```
[the remark this answers is not in the document]
Comment (satyam-harness) [reply src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/ConfigureWebhookNotifications.tsx]: addressed

File: src/modules/20-rbac/interfaces/Notifications.ts [hunk 3]
```diff
@@ -9,12 +9,13 @@ export enum NotificationType {
...
   PagerDuty = 'PagerDuty',
-  MsTeams = 'MsTeams'
+  MsTeams = 'MsTeams',
+  Webhook = 'Webhook'
 }
 
 interface NotificationConfiguration {
   type: NotificationType
-  userGroups: string[]
+  userGroups?: string[]
```
[the remark this answers is not in the document]
Comment (satyam-harness) [reply src/modules/20-rbac/interfaces/Notifications.ts]: created a separate interface for webhookNotificationConfiguration, no need to make userGroups optional

File: src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureMSTeamsNotifications/ConfigureMSTeamsNotifications.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureMSTeamsNotifications/ConfigureMSTeamsNotifications.tsx]: created a separate interface for webhookNotificationConfiguration, no need to make userGroups optional

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/TestWebhookNotification.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      const success = await onClick()
+      if (!success) return
+    }
+    try {
+      setTestStatus(TestStatus.INIT)
+      const resp = await testNotificationSetting({
+        accountId,
+        type: 'WEBHOOK',
+        recipient: testData.webhookUrl,
+        notificationId: 'asd'
```
Comment (YounixIO) [inline src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/TestWebhookNotification.tsx:42]: Hardcoded ID ?

Comment (YounixIO) [reply src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/TestWebhookNotification.tsx:42]: Please add relevant comments around hardcoded values if they are mandatory

Comment (satyam-harness) [reply src/modules/20-rbac/modals/ConfigureNotificationsModal/views/ConfigureWebhookNotifications/TestWebhookNotification.tsx]: comment added

**Review by vikrant-harness — DISMISSED** [review]

**Review by Eric761 — DISMISSED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Comment by bhavya-sinha** [comment]
@satyam-harness Eslint Warning is failing. Please fix it.

**Review by YounixIO — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Pull request merged**